tests/benchdnn/pool/ref_pool.cpp

```diff
@@ -38,7 +38,7 @@ void compute_ref_fwd(const prb_t *prb, const args_t &args) {
         // XXX: this is a hack to let tests with padded area to pass for bf16
         // dt due to the library initialize values with -max_dt, but not -INF.
         float max_value = lowest_dt(prb->dst_dt());
-        if (is_nvidia_gpu()) max_value = lowest_dt(prb->src_dt());
+        if (is_gpu()) max_value = lowest_dt(prb->src_dt());
```

Comment:
Based on the examples shared, I don't see why this is needed. Could you, please, elaborate on this one more?

Reply:
Sorry, I forgot to add the corresponding output about this. The problem is the same we were having with AMD and nVidia cases where the initialization of the values when the max alg is used was causing wrong results. Below you find a couple of examples.

```
./benchdnn --pool --engine=gpu --dir=FWD_I --dt=f32:u8 --tag=axb --attr-post-ops=add:f32:per_oc ic64iw32ow16kw3sw2pw0
[  22][DST][0:1:6] exp_f32:           1 exp:           1 got:           0 diff:       1 rdiff:       1
[  31][DST][0:1:15] exp_f32:           1 exp:           1 got:           0 diff:       1 rdiff:       1
[  66][DST][0:4:2] exp_f32:           5 exp:           5 got:           0 diff:       5 rdiff:       1
[  69][DST][0:4:5] exp_f32:           5 exp:           5 got:           0 diff:       5 rdiff:       1
[  70][DST][0:4:6] exp_f32:           5 exp:           5 got:           0 diff:       5 rdiff:       1
[  71][DST][0:4:7] exp_f32:           5 exp:           5 got:           0 diff:       5 rdiff:       1
[  75][DST][0:4:11] exp_f32:           5 exp:           5 got:           0 diff:       5 rdiff:       1
[  82][DST][0:5:2] exp_f32:           8 exp:           8 got:           0 diff:       8 rdiff:       1
[  92][DST][0:5:12] exp_f32:           8 exp:           8 got:           0 diff:       8 rdiff:       1
[  93][DST][0:5:13] exp_f32:           8 exp:           8 got:           0 diff:       8 rdiff:       1
[COMPARE_STATS][DST]: trh=0 err_max_diff:      16 err_max_rdiff:       1 all_max_diff:      16 all_max_rdiff:       1
0:FAILED (errors:253 total:2048) __REPRO: --pool --engine=gpu --dir=FWD_I --dt=f32:u8 --tag=axb --attr-post-ops=add:f32:per_oc ic64iw32ow16kw3sw2pw0
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total: 0.33s; fill: 0.00s (1%); compute_ref: 0.00s (1%); compare: 0.00s (1%);
TBB Warning: Leaked 2 observer_proxy objects
```

```bash
./benchdnn --pool --engine=gpu --dir=FWD_I --dt=f32:s8 --tag=axb --attr-post-ops=add:f32:per_oc ic64iw32ow16kw3sw2pw0
[  22][DST][0:1:6] exp_f32:        -127 exp:        -127 got:        -128 diff:       1 rdiff:0.00787402
[  31][DST][0:1:15] exp_f32:        -127 exp:        -127 got:        -128 diff:       1 rdiff:0.00787402
[  69][DST][0:4:5] exp_f32:        -123 exp:        -123 got:        -128 diff:       5 rdiff:0.0406504
[  70][DST][0:4:6] exp_f32:        -123 exp:        -123 got:        -128 diff:       5 rdiff:0.0406504
[  71][DST][0:4:7] exp_f32:        -123 exp:        -123 got:        -128 diff:       5 rdiff:0.0406504
[  75][DST][0:4:11] exp_f32:        -123 exp:        -123 got:        -128 diff:       5 rdiff:0.0406504
[  82][DST][0:5:2] exp_f32:        -120 exp:        -120 got:        -128 diff:       8 rdiff:0.0666667
[  92][DST][0:5:12] exp_f32:        -120 exp:        -120 got:        -128 diff:       8 rdiff:0.0666667
[  93][DST][0:5:13] exp_f32:        -120 exp:        -120 got:        -128 diff:       8 rdiff:0.0666667
[  94][DST][0:5:14] exp_f32:        -120 exp:        -120 got:        -128 diff:       8 rdiff:0.0666667
[COMPARE_STATS][DST]: trh=0 err_max_diff:      16 err_max_rdiff:0.142857 all_max_diff:      16 all_max_rdiff:0.142857
0:FAILED (errors:217 total:2048) __REPRO: --pool --engine=gpu --dir=FWD_I --dt=f32:s8 --tag=axb --attr-post-ops=add:f32:per_oc ic64iw32ow16kw3sw2pw0
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total: 0.40s; fill: 0.01s (2%); compute_ref: 0.00s (1%); compare: 0.00s (1%);
TBB Warning: Leaked 2 observer_proxy objects
```
The issue also happens on Intel GPUs, that is why I am changing the initialization values if the device is a GPU.

Thanks for the feedback @dzarukin

Reply:
These are not the cases this change targets. The comment clearly states it's for pX >= kX. It means either the comment somehow is outdated, or an issue might be somewhere else.
As OpenCL backend didn't require such changes, I'd expect the latter.